src/libs/ajax/TerraDataRepo.js

```diff
@@ -1,12 +1,31 @@
 import { Config } from '../config';
 import axios from 'axios';
-
+import {partition} from '../utils';
 
 export const TerraDataRepo = {
   listSnapshotsByDatasetIds: async (identifiers) => {
-    // Note that TDR is expecting dataset identifiers, not dataset ids
-    const url = `${await Config.getTdrApiUrl()}/api/repository/v1/snapshots?duosDatasetIds=${identifiers.join('&duosDatasetIds=')}`;
-    const res = await axios.get(url, Config.authOpts());
-    return res.data;
+    // We can hit max url length (2048) with many identifiers so partition them to a reasonable number and recombine
+    // back into the EnumerateSnapshotModel schema: https://data.terra.bio/swagger-ui.html#/snapshots/enumerateSnapshots
+    // DUOS ID + param (?duosDatasetIds=DUOS-000852): 27 chars
+    // ~70 should a safe default at 1890 characters
+    const partitionedIdentifiers = partition(identifiers, 70);
+    let enumerateSnapshotModel = {
+      total: 0,
+      filteredTotal: 0,
+      items: [],
+      roleMap: {},
+      errors: []
+    };
+    for await (const sublist of partitionedIdentifiers) {
```

Comment:
This makes each async call synchronously. Since each call is independent, they could all be made in parallel which should make this code faster. Would it be hard to change this to use a `Promise.all()` construct instead?

Reply:
> This makes each async call synchronously. Since each call is independent, they could all be made in parallel which should make this code faster. Would it be hard to change this to use a `Promise.all()` construct instead?

Yes, I think that shouldn't be too much of a lift

Reply:
@pshapiro4broad - PTAL, the requests are now all executed in parallel.